server/app/tasks/DatabaseSeedTask.java

```diff
@@ -0,0 +1,162 @@
+package tasks;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.ImmutableSet;
+import io.ebean.DB;
+import io.ebean.Database;
+import io.ebean.SerializableConflictException;
+import io.ebean.Transaction;
+import io.ebean.TxScope;
+import io.ebean.annotation.TxIsolation;
+import java.util.Optional;
+import java.util.stream.Collectors;
+import javax.inject.Inject;
+import javax.inject.Provider;
+import javax.persistence.NonUniqueResultException;
+import javax.persistence.RollbackException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import play.i18n.Lang;
+import repository.VersionRepository;
+import services.CiviFormError;
+import services.ErrorAnd;
+import services.LocalizedStrings;
+import services.question.QuestionService;
+import services.question.types.DateQuestionDefinition;
+import services.question.types.NameQuestionDefinition;
+import services.question.types.QuestionDefinition;
+
+/**
+ * Task for seeding the database. All of its work is done in the constructor, which is its only
+ * public method.
+ *
+ * <p>Logic for seeding different resources should be factored into separate methods for clarity.
+ *
+ * <p>To avoid attempting to insert duplicate resources it uses a thread-local database transaction
+ * with a transaction scope of SERIALIZABLE. If the transaction fails it retries up to {@code
+ * MAX_RETRIES} times.
+ */
+public final class DatabaseSeedTask {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(DatabaseSeedTask.class);
+  private static final int MAX_RETRIES = 10;
+  private static final ImmutableList<QuestionDefinition> CANONICAL_QUESTIONS =
+      ImmutableList.of(
+          new NameQuestionDefinition(
```

Comment:
Not blocking and way outside the scope of this PR: Is there a builder pattern established for this that we could leverage rather than needing to comment-document each of the fields below?

Reply:
[There is](https://github.com/seattle-uat/civiform/blob/main/server/app/services/question/types/QuestionDefinitionBuilder.java#L18). I decided against using it because `QuestionDefinitionBuilder#build()` requires handling a `UnsupportedQuestionTypeException` which would make for messier code in this class IMO. Thoughts?

Reply:
done, added a new method `unsafeBuild` to get around error handling